variables.tf

```diff
@@ -144,6 +144,27 @@ variable "managed_rules" {
   default = {}
 }
 
+variable "recording_mode" {
+  description = <<-DOC
+    The mode for AWS Config to record configuration changes. Possible values are:
```

Comment:
We could create a `validation` block for these values in this variable, but not required.

Reply:
Adding validation client side means we need to publish new versions as the API changes. My preference is to provide clear examples, optional attributes, and let the APIs do the validation. But, I am open to adding it if you feel it's necessary.